### ISSUE TITLE
Improve readability of the pandoc command in the pdf generation script

### ISSUE THREAD

**Issue by Mister00X** [siduction/sidu-manual#46]
While solving the language issue I realized that this line https://github.com/siduction/sidu-manual/blob/283633c64daaeaf2d0a1dbc6058bcb024d6f7709/development/10-generate-pdf-manual.sh#L108 is especially hard to read because it's so long.

Maybe it would be possible to change it to one option per line.
I imagine something like this:
```sh
pandoc -N --toc-depth=4 \
       -p --pdf-engine=xelatex \
       --listings \
       .....
```
This would make it more readable and probably better `grep`-able.

Thoughts on this?

**Comment by ak-li** [comment]
Good idea. I did it.

~~~
pandoc \
     -N --toc --toc-depth=4 \
     -p --pdf-engine=xelatex \
     -H $header \
     --listings \
     --resource-path=$searchpath \
     --highlight-style tango \      <--   without '=' , is this right?
     -V lang=$language \
     -V papersize=$paper \
     -V hmargin=$hmargin \
     -V vmargin=$vmargin \
     -V linkcolor=$linkcolor \
     -V geometry:margin="$margin" \
     -V mainfont="$mainfont" \
     -V sansfont="$sansfont" \
     -V monofont="$monofont" \
     -V geometry=$geometry \
     -V alignment=$alignment \
     -V columns=$columns \
     -V fontsize=$fontsize \
     -V nohyphenation=$nohyphenation $LISTE \
     -o ../data/$langcode/pdf/siduction-manual_$langcode.pdf
~~~

**Comment by Mister00X** [comment]
> --highlight-style tango \ Is this ok? Without '=' ?

Good catch. According to https://pandoc.org/MANUAL.html#option--highlight-style it is indeed `--highlight-style=<STYLENAME>`.

Btw did githubs markdown eat away all the `\` in your comment?

**Comment by ak-li** [comment]
I'll change to `--highlight-style=<STYLENAME>`

> Btw did githubs markdown eat away all the `\` in your comment?

Changed layout of previous comment to show the backslash.

**Comment by Mister00X** [comment]
> Changed layout of previous comment to show the backslash.

Ah thanks :+1: I just wanted to avoid any misunderstandings.

**Comment by ak-li** [comment]
Now I tested `--highlight-style=tango`

The option has no effect on the text. 
Not with other styles, not with different languages and not if it is moved after the -o option (I read somewhere).
Only the '#' causes italic text.

**Comment by Mister00X** [comment]
Am 16. April 2022 15:37:04 MESZ schrieb ak-li ***@***.***>:
>Now I tested `--highlight-style=tango`
>
>The option has no effect on the text. 
>Not with other styles, not with different languages and not if it is moved after the -o option (I read somewhere).
>Only the '#' causes italic text.
>


Interesting. To be honest I'm not entirely sure why it does that. Maybe it has no highlight-style for shell-syntax or it conflicts with listings. I should really rtfm of pandoc :)

**Comment by ak-li** [comment]
I searched and tested for several hours, but no acceptable result.
In `11-helpfile-pdf-manual-header.tex` there are definitions for code highlight starting at line 79.
If they are activated, we get only partially reasonable output. The biggest flaw is the use of the hash (#). Highlighted is the one comment, we use it to mark the root terminal. An unsolvable conflict.

**Comment by Mister00X** [comment]
@ak-li Did you also tested removing the `--listings` option and leaving the  `--highlight-style=tango` option as is?

What do you think is the best solution?
Leave `--highlight-style=tango` as is, comment the option out or remove it entirely?

**Comment by ak-li** [comment]
@Mister00X 
> Did you also tested removing the `--listings` option and leaving the `--highlight-style=tango` option as is?

Yes:
- remove only `--listings`  in `10-generate-pdf-manual.sh` -> compiler error `! Undefined control sequence. <recently read> \lstset `
  - additionally comment `\lstset` in `11-helpfile-pdf-manual-header.tex` -> other compiler error
-  remove only `--highlight-style=tango`  in `10-generate-pdf-manual.sh` -> no changes
- made changes as followed  in `11-helpfile-pdf-manual-header.tex` -> colour is on and comment (#) is not italic
  
  ~~~
  \lstset{
    %basicstyle=\ttfamily\footnotesize,     -> comment out
    [...]
    % code highlight
    language=bash,
    basicstyle=\color[RGB]{40,1,196}\ttfamily\footnotesize,   -> insert "\ttfamily\footnotesize"
    %keywordstyle=\color[rgb]{0.13,0.29,0.53}\bfseries,
    keywordstyle=\color[RGB]{42,161,152}\bfseries,
    %stringstyle=\color[rgb]{0.31,0.60,0.02},
    stringstyle=\color[RGB]{28,139,210},
    numberstyle=\color[RGB]{108,113,196},
    identifierstyle=\color[RGB]{0,0,74},
    %commentstyle=\color[rgb]{0.56,0.35,0.01}\itshape,
    %commentstyle=\color[RGB]{0,85,0},
    commentstyle=\color[RGB]{0,0,74},    -> colour changed and italic removed
    %rulecolor=\color{red},
  ~~~

Please wait a bit. I search for bg-color inline code.

But this isn't solved:

> The biggest flaw is the use of the hash (#). [...], we use it to mark the root terminal.

**Comment by ak-li** [comment]
@Mister00X 
> Please wait a bit. I search for bg-color inline code.

There is no useful way to declare the background color for inline code in the header when using "--listings". This is only possible with LaTex code in the md document. :-((

Now the change I did:

> basicstyle=\color{red!35!black}\ttfamily\normalsize,
> backgroundcolor=\color{gray!10!white},

`basicstyle` only once with all options needed. Color in human readable code. Font size normal. Easily readable in text. Recognizable as code by the same color as in the block. By the color not to be confused with link.

> language=bash,
> keywordstyle=\normalsize,
> stringstyle=\normalsize,
> numberstyle=\normalsize,
> identifierstyle=\normalsize,
> commentstyle=\normalsize,

No highlighting and `commentstyle=\normalsize` to prevent cursive text markup.

Here a test file: [siduction-manual_en.pdf](https://github.com/siduction/sidu-manual/files/8502212/siduction-manual_en.pdf)

**Comment by ak-li** [comment]
Another test file with font size "\small" instead of "\normalsize".
[siduction-manual_en.pdf](https://github.com/siduction/sidu-manual/files/8507817/siduction-manual_en.pdf)

**Comment by Mister00X** [comment]
@ak-li Imho the version with `\small` looks better to me. Also some code blocks fit into one line thanks to that.  

Btw (and I am asking this because I'm completely clueless about that topic) could the reddish colour for the code lead to accessibility problems with people who suffer from colour-blindness?

**Comment by ak-li** [comment]
@Mister00X 
> Imho the version with `\small` looks better to me.

Then we use `\small`.

> problems with people who suffer from colour-blindness?

+ **Color blindness**: No colors can be perceived, only contrasts light-dark (0,001% men and women).
  Inline text: Differentiation exclusively by font size and style. May be we should use `\bfseries` (bold) without color.
  Block code is easily readable through the background.

+ **Color vision deficiency**:
  <ins>L-System (without red)</ins>: 1% men, 0,02% women [green -> yellow/lightgrey and red -> blue/darkgrey]
  <ins>M-System (without green)</ins>: 1,1% men, 0,01% women [green -> grey/yellow and red -> blue/grey/yellow]
  Good perception of blue and yellow.
  These two restrictions are relevant to the manual. The colors that are distinguishable for both are green and slightly worse orange.
  <ins>S-System (without blue)</ins>: 0,002% men, 0,001% women [green,yellow,blue -> turquoise/red/grey]
  Good perception of red and green (becomes mostly turquoise).
  Yellow appears pink, green become turquoise with gray and orange become red with gray.

+ **Warning bg-color**
  I changed it to yellow. This makes the warning stand out strongly for everyone.

To generate the pattern files I used the program of this website: [ColorBlindnessSim](http://lpetrich.org/Science/ColorBlindnessSim/ColorBlindnessSim.html).
The test file from above [siduction-manual_en.pdf](https://github.com/siduction/sidu-manual/files/8507817/siduction-manual_en.pdf) becomes mainly gray for the visually impaired.

If we use colors, green would be my recommendation.  Then we should add a color vision explanation under *1Wellcome* > *1.1.1 General*. If we don't use colors, bold text would be useful for code. But that doesn't look so good in code blocks.

The pattern files:
[colortest_green-no_blue.pdf](https://github.com/siduction/sidu-manual/files/8515399/colortest_green-no_blue.pdf)
[colortest_green-no_green.pdf](https://github.com/siduction/sidu-manual/files/8515400/colortest_green-no_green.pdf)
[colortest_green-no_red.pdf](https://github.com/siduction/sidu-manual/files/8515401/colortest_green-no_red.pdf)
[colortest_green-normal.pdf](https://github.com/siduction/sidu-manual/files/8515403/colortest_green-normal.pdf)
[colortest_orange-no_blue.pdf](https://github.com/siduction/sidu-manual/files/8515404/colortest_orange-no_blue.pdf)
[colortest_orange-no_green.pdf](https://github.com/siduction/sidu-manual/files/8515405/colortest_orange-no_green.pdf)
[colortest_orange-no_red.pdf](https://github.com/siduction/sidu-manual/files/8515406/colortest_orange-no_red.pdf)
[colortest_orange-normal.pdf](https://github.com/siduction/sidu-manual/files/8515407/colortest_orange-normal.pdf)

**Comment by Mister00X** [comment]
> * Inline text: Differentiation exclusively by font size and style. May be we should use `\bfseries` (bold) without color.

As someone who is kinda used to it, I believe that maybe the switch to a monospaced font from a sans serif font could be clear enough. (That's entirely subjective though.)

> If we use colors, green would be my recommendation. Then we should add a color vision explanation under _1Wellcome_ > _1.1.1 General_. If we don't use colors, bold text would be useful for code. But that doesn't look so good in code blocks.

I am very much for the use of colours. Regarding the colour vision explanation do you want to generate two versions of the manual one special version for colour blind people and one regular version? Or did I misunderstood this and you want to produce only a colour blind friendly version and add an explanation nevertheless?

**Comment by ak-li** [comment]
> and you want to produce only a colour blind friendly version and add an explanation nevertheless?

That's exactly how I imagined it.
Because the colored version is also attractive for normal-sighted people. The explanation rather serves to justify the relatively bright color.

**Comment by Mister00X** [comment]
I see. I'm fine with that :+1:

**Comment by ak-li** [comment]
Which color do you think is better?
If we agree, I would transfer the suggestion to a new issue and ask the other members.

**Comment by Mister00X** [comment]
Personally I like the orange version more, I feel like green is kinda too much (like too eye catchy). The orange version is to me more visually pleasing.

**Comment by ak-li** [comment]
Also OK
I start a new issue and put both variants to the vote.
Then I enter both color variants into `11-helpfile-pdf-manual-header.tex` and close this issue.

**Comment by ak-li** [comment]
closed with commit https://github.com/siduction/sidu-manual/commit/ec396c08d9967660c40e7c6966dd5b708b7f1014
